packages/features/bookings/lib/handleNewBooking/ensureAvailableUsers.ts

```diff
@@ -121,6 +122,81 @@ const _ensureAvailableUsers = async (
     },
   });
 
+  if (eventType.restrictionScheduleId) {
+    try {
+      const restrictionSchedule = await prisma.schedule.findUnique({
+        where: { id: eventType.restrictionScheduleId },
+        select: {
+          id: true,
+          timeZone: true,
+          availability: {
+            select: {
+              days: true,
+              startTime: true,
+              endTime: true,
+              date: true,
+            },
+          },
+        },
+      });
+      if (restrictionSchedule) {
```

Comment:
We could reduce nesting and add log if restriction schedule was not found because now it is silently ignored:
```
if (!restrictionSchedule) {
     loggerWithEventDetails.error(
            `No restriction schedule with id={eventType.restrictionScheduleId} found.`
       );
       return;
}
if (!eventType.useBookerTimezone && !restrictionSchedule?.timeZone) {
    loggerWithEventDetails.error(
            `No timezone set for restriction schedule.`,
            piiFreeInputDataForLogging
       );
}

const restrictionTimezone = eventType.useBookerTimezone
          ? input.timeZone
          : restrictionSchedule.timeZone ?? undefined
```

this way we also log that timezone is not set for restriction schedule if it is not found in the first place